67_RayQueryGeometry/main.cpp

```diff
@@ -0,0 +1,1010 @@
+					.srcBaseLayer = 0,
+					.dstBaseLayer = 0,
+					.srcMipLevel = 0,
+					.dstMipLevel = 0,
+					.aspectMask = IGPUImage::E_ASPECT_FLAGS::EAF_COLOR_BIT
+				}};
+				
+				auto srcImg = outHDRImageViews[m_currentImageAcquire.imageIndex]->getCreationParameters().image.get();
+				auto scRes = static_cast<CDefaultSwapchainFramebuffers*>(m_surface->getSwapchainResources());
+				auto dstImg = scRes->getImage(m_currentImageAcquire.imageIndex);
+
+				cmdbuf->blitImage(srcImg, IImage::LAYOUT::TRANSFER_SRC_OPTIMAL, dstImg, IImage::LAYOUT::TRANSFER_DST_OPTIMAL, regions, ISampler::ETF_NEAREST);
+			}
+
+			// TODO: transition to present
+			{
+				IGPUCommandBuffer::SPipelineBarrierDependencyInfo::image_barrier_t imageBarriers[2];
+				imageBarriers[1].barrier = {
+				   .dep = {
+					   .srcStageMask = PIPELINE_STAGE_FLAGS::BLIT_BIT,
+					   .srcAccessMask = ACCESS_FLAGS::TRANSFER_WRITE_BIT,
+					   .dstStageMask = PIPELINE_STAGE_FLAGS::NONE,
+					   .dstAccessMask = ACCESS_FLAGS::NONE
+					}
+				};
+				imageBarriers[1].image = m_surface->getSwapchainResources()->getImage(m_currentImageAcquire.imageIndex);
+				imageBarriers[1].subresourceRange = {
+					.aspectMask = IImage::EAF_COLOR_BIT,
+					.baseMipLevel = 0u,
+					.levelCount = 1u,
+					.baseArrayLayer = 0u,
+					.layerCount = 1u
+				};
+				imageBarriers[1].oldLayout = IImage::LAYOUT::TRANSFER_DST_OPTIMAL;
+				imageBarriers[1].newLayout = IImage::LAYOUT::PRESENT_SRC;
+
+				cmdbuf->pipelineBarrier(E_DEPENDENCY_FLAGS::EDF_NONE, { .imgBarriers = imageBarriers });
+			}
+
+			{
+				const IQueue::SSubmitInfo::SSemaphoreInfo rendered[] =
+				{
+					{
+						.semaphore = m_semaphore.get(),
+						.value = ++m_realFrameIx,
+						.stageMask = PIPELINE_STAGE_FLAGS::ALL_GRAPHICS_BITS
+					}
+				};
+				{
+					{
+						const IQueue::SSubmitInfo::SCommandBufferInfo commandBuffers[] =
+						{
+							{.cmdbuf = cmdbuf }
+						};
+
+						const IQueue::SSubmitInfo::SSemaphoreInfo acquired[] =
+						{
+							{
+								.semaphore = m_currentImageAcquire.semaphore,
+								.value = m_currentImageAcquire.acquireCount,
+								.stageMask = PIPELINE_STAGE_FLAGS::NONE
+							}
+						};
+						const IQueue::SSubmitInfo infos[] =
+						{
+							{
+								.waitSemaphores = acquired,
+								.commandBuffers = commandBuffers,
+								.signalSemaphores = rendered
+							}
+						};
+
+						if (queue->submit(infos) == IQueue::RESULT::SUCCESS)
+						{
+							const nbl::video::ISemaphore::SWaitInfo waitInfos[] =
+							{ {
+								.semaphore = m_semaphore.get(),
+								.value = m_realFrameIx
+							} };
+
+							m_device->blockForSemaphores(waitInfos); // this is not solution, quick wa to not throw validation errors
+						}
+						else
+							--m_realFrameIx;
+					}
+				}
+
+				std::string caption = "[Nabla Engine] Geometry Creator";
+				{
+					caption += ", displaying [" + std::string(object.meta.name.data()) + "]";
+					m_window->setCaption(caption);
+				}
```

Comment:
take the choice of what to display away, display all geometries in a line side-by-side (one TLAS, multiple BLASes - one per geo)

Reply:
done?